sys/log/full/src/log_fcb.c

```diff
@@ -532,19 +532,30 @@ log_fcb_registered(struct log *log)
 {
 #if MYNEWT_VAL(LOG_STORAGE_WATERMARK)
     struct fcb_log *fl;
+#if !MYNEWT_VAL(LOG_REBOOT_INIT_WATERMARK_UNK)
     struct fcb *fcb;
     struct fcb_entry loc;
+#endif
 
     fl = (struct fcb_log *)log->l_arg;
+
+#if !MYNEWT_VAL(LOG_REBOOT_INIT_WATERMARK_UNK)
     fcb = &fl->fl_fcb;
 
     /* Set watermark to first element */
     memset(&loc, 0, sizeof(loc));
+#endif
+
+#if MYNEWT_VAL(LOG_REBOOT_INIT_WATERMARK_UNK)
+    /* Initialize watermark to designated unknown value*/
+    fl->fl_watermark_off = (uint32_t)MYNEWT_VAL(LOG_INIT_WATERMARK_VAL);
+#else
     if (fcb_getnext(fcb, &loc)) {
         fl->fl_watermark_off = loc.fe_area->fa_off + loc.fe_elem_off;
     } else {
         fl->fl_watermark_off = fcb->f_oldest->fa_off;
     }
+#endif
 #endif
     return 0;
```

Comment:
This function (`log_fcb_registered()`) is starting to get a bit confusing with all the `#if` / `#endif` directives.  Can you combine the bottom half into a single `#if` / `#else`?